articles/110_interface_definition.md

```diff
@@ -126,11 +126,88 @@ Both file names must use an upper camel case name and only consist of alphanumer
 Field names must be lowercase alphanumeric characters with underscores for separating words.
 They must start with an alphabetic character, they must not end with an underscore and never have two consecutive underscores.
 
-### Naming of constants
+#### Naming Conflicts
 
-Constant names must be uppercase alphanumeric characters with underscores for separating words.
+When defining field names it has been recommended to avoid language keywords and other reserved symbols.
+However since we cannot be guaranteed to know all conflicts of potential future languages or implementations we must have a way to systematically deal with these conflicts without disrupting existing code bases and implementations.
+To that end we will require that language bindings provide a reproducable name mangling mapping that will prevent conflicts with reserved keywords or other conflicts.
+
+##### Background
+
+This has come up specifically as we're adding support for Windows for ROS2.
+`winnt.h` defines several macros that conflict with existing enumerations.
+It is also expected to happen as support for new languages are added.
+We cannot know all the potential future keywords and restrictions from a language which is selected to add support for in the future so we must have a generic solution which will allow future languages to be added without disrupting the existing usages.
+
+##### Language Generator Requirements
+
+In the case that a new language is added and there is a detected conflicting name for a field, the language specific generator is expected to implement a basic deconfliction mangling of the symbol to avoid the conflict.
+
+Each generator will define the mangling procedure and provide a list of symbols for which the procedure will be applied.
+This list should be all known keywords or otherwise conflicting symbols.
+
+##### Example Deconfliction Policy
+
+A example of a simple name mangling policy would be to append a trailing underscore for conflicting symbols.
+So that would mean that `FOO` if declared as a conflicting symbol would be used in generated code as `FOO_`.
+
+Clearly this would collide if someone defined `class` and `class_`, so slightly more unique string is recommended for clarity and uniqueness.
+The mangling should be designed so that accidental collisions between names and mangled names cannot happen.
+One easy way to do this would be to a language specific valid character not valid in the standard field name specification.
```

Comment:
The [current state of ros2 idl restrictions](https://github.com/ros2/design/pull/172/files#diff-117bf5c2128f31db6fc024e9d482b563R127) prevents fields from being defined with a trailing underscore. If we aren't planning to relax that (I think it wasn't imposed in ros1), then perhaps a trailing underscore can always used for mangling, without any potential for accidental collisions?